docs/Training-Generalization-Learning.md

```diff
@@ -0,0 +1,123 @@
+# Training Generalized Reinforcement Learning agents
+
+Reinforcement learning has a rather unique setup as opposed to supervised and
+unsupervised learning. Agents here are trained and tested on the same exact 
+environment, which is analogous to a model being trained and tested on an 
+identical dataset in supervised learning! This setting results in overfitting; 
+the inability of the agent to generalize to slight tweaks or variations in the 
+environment. This is problematic in instances when environments are randomly 
+instantiated with varying properties. To make agents more robust, we train an 
```

Comment:
Would rephrase: "To make agents more robust, one approach is to train over multiple variations of the environment."